lib/jdomodels/src/test/java/org/sagebionetworks/repo/model/dbo/file/download/v2/DownloadListDaoImplTest.java

```diff
@@ -0,0 +1,363 @@
+package org.sagebionetworks.repo.model.dbo.file.download.v2;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.sagebionetworks.repo.model.UserGroup;
+import org.sagebionetworks.repo.model.UserGroupDAO;
+import org.sagebionetworks.repo.model.download.IdAndVersion;
+import org.sagebionetworks.repo.model.jdo.KeyFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.google.common.collect.Lists;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(locations = { "classpath:jdomodels-test-context.xml" })
+public class DownloadListDaoImplTest {
+
+	@Autowired
+	private UserGroupDAO userGroupDao;
+	@Autowired
+	private DownloadListDAO downloadListDao;
+
+	Long userOneIdLong;
+	String userOneId;
+
+	Long userTwoIdLong;
+	String userTwoId;
+
+	List<IdAndVersion> idsWithVersions;
+	List<IdAndVersion> idsWithoutVersions;
+
+	@BeforeEach
+	public void before() {
+		downloadListDao.truncateAllData();
+		UserGroup ug = new UserGroup();
+		ug.setCreationDate(new Date(System.currentTimeMillis()));
+		ug.setIsIndividual(true);
+		userOneIdLong = userGroupDao.create(ug);
+		userOneId = "" + userOneIdLong;
+		// second user
+		ug = new UserGroup();
+		ug.setCreationDate(new Date(System.currentTimeMillis()));
+		ug.setIsIndividual(true);
+		userTwoIdLong = userGroupDao.create(ug);
+		userTwoId = "" + userTwoIdLong;
+
+		int numberOfIds = 4;
+		idsWithVersions = new ArrayList<IdAndVersion>(numberOfIds);
+		idsWithoutVersions = new ArrayList<IdAndVersion>(numberOfIds);
+		for (int i = 0; i < numberOfIds; i++) {
+
+			IdAndVersion idWithVersion = new IdAndVersion();
+			idWithVersion.setEntityId("syn" + i);
+			idWithVersion.setVersionNumber(new Long(i));
+			idsWithVersions.add(idWithVersion);
+
+			IdAndVersion idWithoutVersion = new IdAndVersion();
+			idWithoutVersion.setEntityId("syn" + (i + numberOfIds + 1));
+			idWithoutVersion.setVersionNumber(null);
+			idsWithoutVersions.add(idWithoutVersion);
+		}
+	}
+
+	@AfterEach
+	public void after() {
+		if (userOneId != null) {
+			userGroupDao.delete(userOneId);
+
+		}
+		if (userTwoId != null) {
+			userGroupDao.delete(userTwoId);
+		}
+		downloadListDao.truncateAllData();
+	}
+
+	@Test
+	public void testAddBatchOfFilesToDownloadListWithNullUserId() {
+		List<IdAndVersion> batch = Lists.newArrayList(idsWithVersions.get(0), idsWithoutVersions.get(0));
```

Comment:
We can avoid using guava, e.g. Arrays.asList()

Reply:
It is not the same.  The results of Arrays.asList() are "fixed-size".  If you try to add to the resulting list you get an UnsupportedOperationException.  Basically, the results are as limited as an actual array.